static/js/hero.js

```diff
@@ -46,59 +46,52 @@ $(document).ready(function() {
   // its existence before we try anything.
   $("#actionButton").on("click", function(event) {
     event.preventDefault();
-
-    const hlsAboutVideo = $("#tv-light-box-video").get(0);
-    const aboutVideoYoutube = $("#tv-yt-light-box-video")
-      .find("iframe")
-      .get(0);
-
-    if (hlsAboutVideo) {
-      showLightBox();
-      hlsAboutVideo.play();
-    } else if (aboutVideoYoutube) {
-      showLightBox();
-      aboutVideoYoutube.contentWindow.postMessage(
-        JSON.stringify({
-          event: "command",
-          func:  "playVideo"
-        }),
-        "https://www.youtube.com"
-      );
+    const youtubeVideoSrc = $("#tv-light-box-yt-video").attr("data-href");
+    const hlsAboutVideoSrc = $("video#tv-light-box-video");
+    let backgroundVideo = null;
+    if (youtubeVideoSrc) {
+      $.fancybox.open({
+        src:        youtubeVideoSrc,
+        beforeShow: function() {
+          backgroundVideo = $("#background-video").get(0);
+        },
+        beforeLoad: function() {
+          pauseBackgroundVideo(backgroundVideo);
+        },
+        afterClose: function() {
+          resumeBackgroundVideo(backgroundVideo);
+        }
+      });
+    } else if (hlsAboutVideoSrc) {
+      $.fancybox.open({
+        content:    hlsAboutVideoSrc,
+        type:       "html",
+        beforeShow: function() {
+          backgroundVideo = $("#background-video").get(0);
+        },
+        beforeLoad: function() {
+          pauseBackgroundVideo(backgroundVideo);
+        },
+        afterClose: function() {
+          resumeBackgroundVideo(backgroundVideo);
+        }
+      });
```

Comment:
I think this repetition can be cut down and the onClick logic can be separated in a cleaner way:

```javascript
const backgroundVideoSel = "#background-video";
const promoVideoSel = "#promo-video";

// ...

function openVideoLightbox() {
  const youtubeVideoSrc = $("#tv-light-box-yt-video").attr("data-href");
  const hlsAboutVideoSrc = $("video#tv-light-box-video");
  if (!youtubeVideoSrc && !hlsAboutVideoSrc) {
    console.error("We do not have any supported video elements available."); // eslint-disable-line no-console
    return;
  }

  let backgroundVideo = null;
  const fancyboxArgs = $.extend(
    {
      beforeShow: function() {
        backgroundVideo = $(backgroundVideoSel).get(0);
      },
      beforeLoad: function() {
        pauseBackgroundVideo(backgroundVideo);
      },
      afterClose: function() {
        resumeBackgroundVideo(backgroundVideo);
      }
    },
    youtubeVideoSrc ? {
      src: youtubeVideoSrc
    } : {
      content: hlsAboutVideoSrc,
      type:    "html"
    }
  );
  $.fancybox.open(fancyboxArgs);
}

$(document).ready(function() {
  // Background cover video in header on home page
  configureHlsVideo(backgroundVideoSel, true);

  // Promo video in header on product detail page
  configureHlsVideo(promoVideoSel);

  // The action button is supposed to play a video element in light box.
  // which exists in another section, which is why we need to check for
  // its existence before we try anything.
  $("#actionButton").on("click", function(event) {
    event.preventDefault();
    openVideoLightbox();
  });
});
```

Reply:
Actually for `FancyBox` here, I used `$("video#tv-light-box-video")` to get the whole element instead of index object for `FancyBox`.